server/routes/authenticate.js

```diff
@@ -10,26 +20,34 @@ router.post('/', (req, res) => {
   const reqPassword = req.body.password;
 
   if (reqEmail && reqPassword) {
-    User.getUserByEmail({ email: reqEmail })
+    User.getUserByEmail({ email: reqEmail }, usersModel)
```

Comment:
There are too many callbacks in callbacks here! Separate each task and use `Promise` to chain them up.